BootloaderCommonPkg/Library/Ext23Lib/ExtLib.c

```diff
@@ -270,3 +270,40 @@ ExtFsCloseFile (
   Ext2fsClose (OpenFile);
   FreePool (OpenFile);
 }
+
+/**
+  List directories or files
+
+  @param[in]     FsHandle         file system handle.
+  @param[in]     DirFilePath      directory or file path
+
+  @retval EFI_SUCCESS             list directories of files successfully
+  @retval EFI_UNSUPPORTED         this api is not supported
+  @retval Others                  an error occurs
+
+**/
+EFI_STATUS
+EFIAPI
+ExtFsListDir (
+  IN  EFI_HANDLE                                  FsHandle,
+  IN  CHAR16                                     *DirFilePath
+  )
+{
+  EFI_STATUS              Status;
+  EFI_HANDLE              FileHandle;
+
+  Status = EFI_UNSUPPORTED;
+
+DEBUG_CODE_BEGIN ();
```

Comment:
never mind for the previous comments.
Looks like this command is only available for debug build.
I was thinking it might be better to use a different PCD to control if a command should be supported in full or mini Shell.  In this way, if for any reason, people want to run full shell in release build, it is still possible.

Reply:
I agree to use a PCD to control supported shell commands. Until we finalize supported shell commands, is it okay to keep this for now?

Reply:
OK, let us do a follow up once we have the decision on how to deal with Shell in release mode.